server/src/main/java/org/opensearch/common/bytes/AbstractBytesReference.java

```diff
@@ -82,7 +84,13 @@ public void writeTo(OutputStream os) throws IOException {
 
     @Override
     public String utf8ToString() {
-        return toBytesRef().utf8ToString();
+        BytesRef bytesRef = toBytesRef();
+        final char[] ref = new char[bytesRef.length];
+        final int len = UnicodeUtil.UTF8toUTF16(bytesRef, ref);
+        if (len > MAX_LENGTH) {
+            throw new IllegalStateException("UTF16 String size is " + len + ", should be less than " + MAX_LENGTH);
+        }
+        return new String(ref, 0, len);
```

Comment:
Should explicitly add the UTF_8 charset here.

Reply:
I am not sure, `ref` is already an array of `char` (not `byte`)